packages/block-library/src/list/edit.js

```diff
@@ -14,39 +26,104 @@ export default function ListEdit( {
 	className,
 } ) {
 	const { ordered, values } = attributes;
+	const [ startValue, setStartValue ] = useState( '' );
+	const [ reversed, setReversed ] = useState( false );
+	const [ listType, setListType ] = useState( 'list-type-decimal' );
+
+	const listTypes = [
+		{
+			name: __( 'Decimal' ),
+			type: 'list-type-decimal',
+		},
+		{
+			name: __( 'Lower alpha' ),
+			type: 'list-type-lower-alpha',
+		},
+		{
+			name: __( 'Upper alpha' ),
+			type: 'list-type-upper-alpha',
+		},
+		{
+			name: __( 'Lower roman' ),
+			type: 'list-type-lower-roman',
+		},
+		{
+			name: __( 'Upper roman' ),
+			type: 'list-type-upper-roman',
+		},
+	];
 
 	return (
-		<RichText
-			identifier="values"
-			multiline="li"
-			tagName={ ordered ? 'ol' : 'ul' }
-			onChange={ ( nextValues ) => setAttributes( { values: nextValues } ) }
-			value={ values }
-			wrapperClassName="block-library-list"
-			className={ className }
-			placeholder={ __( 'Write list…' ) }
-			onMerge={ mergeBlocks }
-			unstableOnSplit={
-				insertBlocksAfter ?
-					( before, after, ...blocks ) => {
-						if ( ! blocks.length ) {
-							blocks.push( createBlock( 'core/paragraph' ) );
-						}
+		<Fragment>
+			<RichText
+				identifier="values"
+				multiline="li"
+				tagName={ ordered ? 'ol' : 'ul' }
+				onChange={ ( nextValues ) => setAttributes( { values: nextValues } ) }
+				value={ values }
+				start={ startValue }
+				reversed={ reversed }
+				wrapperClassName="block-library-list"
+				className={ classnames( className, listType ) }
```

Comment:
Why is list type set as a class? The `ol` element support `type` as an attribute in HTML5. https://html.spec.whatwg.org/multipage/grouping-content.html#attr-ol-type

Reply:
I have tried to use the `type` as an attribute here, but I failed. This is because the stylesheet < link rel="stylesheet" id="common-css" href="http://localhost:8888/wp-admin/css/common.css?ver=5.1.1" type="text/css" media="all" > will override the `type` attribute. The stylesheet has set the ordered list type to decimal.
```CSS
ol {
	list-style-type: decimal;
	margin-left: 2em;
}
```

Reply:
Hm, why is that rule there? Can we not simply remove it? I believe `decimal` is the default, so the rule seems redundant to me. @jasmussen Do you have any idea by any chance?

Reply:
There is bleed from the wp-admin, correct, this is what the editing canvas inherits from wp-admin:

<img width="326" alt="Screenshot 2019-04-24 at 11 58 16" src="https://user-images.githubusercontent.com/1204802/56650931-4d4d2e80-6688-11e9-8558-55fe223fadc5.png">

However I do think you're right, decimal _is_ the default for the ol, so it can be removed here. I don't know why it's explicitly set as such in the wp-admin stylesheet, and I'd love for us to look at that separately to scope it better.

Reply:
Thank you @jasmussen!

So there are two options @Jackie6:

* Remove the rule. (Ideal, but requires core change.)
* Set the styles by attribute, not with the aid of a class. With that I mean:

```css
ol[type="a"] {
  list-style-type: lower-alpha;
}
```

Instead of using a class:

```css
.editor-styles-wrapper .block-library-list .list-type-lower-alpha {
  list-style-type: lower-alpha;
}
```

This is a bit cleaner I think.

Reply:
Hi @ellatrix By using the attribute selector, we need to distinguish "a" and "A", because `[type="a"]` is case-insensitive. CSS4 have introduced the [attribute case-sensitivity selector](https://css4-selectors.com/selector/css4/attribute-case-sensitivity/), so we can write code as follows
```CSS
.editor-styles-wrapper .block-library-list ol[type="a" s] {
  list-style-type: lower-alpha;
}

.editor-styles-wrapper .block-library-list ol[type="A" s] {
	list-style-type: upper-alpha;
}
```
However, this new feature is not supported in many browsers including Chrome. You can get browser test results [here](https://css4-selectors.com/browser-selector-test/)
We think we can (1) reduce the five options of list types to three or (2) use the class as before or (3) remove the rule in wp core

Reply:
Ok, that's fine then, in that case you could only use the class in the editor to override the admin stylesheet.

Reply:
The CSS styles have been updated :)